mitodl/open-discussions
# support richtext editing/display in ol-widgets

#### Pre-Flight checklist


#### What are the relevant tickets?
#3707 

#### What's this PR do?
This PR implements rich text editing / display for the RichText widget (`widget_type: markdown`).
- editing is via CKEditor *The old open-discussions rich text / markdown widget is built upon Prosemirror.*
- display is via React Markdown *Same as open-discussions, but using v6 instead of v3*

The automated tests for this are a bit light... CKEditor uses some browser APIs that are not supported in JsDom (e.g., contenteditable) and so doesn't really function in jest.



#### How should this be manually tested?
Go to a field page you have moderator rights to (e.g., by being logged in as a superuser) and:
1. Click the gear -> "Manage Widgets"
2. Add some rich text widgets if you don't have them yet
3. Edit them. Try rich text stuff... lists, links, bold/italics
4. When done editing, click "Done" to save your changes to the widgets
5. Check that look right.
6. Try more editing.

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
See also https://github.com/mitodl/hq/discussions/210.

#### Screenshots (if appropriate)
<img width="593" alt="Screen Shot 2022-08-30 at 12 14 59 PM" src="https://user-images.githubusercontent.com/9010790/187493364-1e136718-b4d7-4087-af87-7f8f72a73f09.png">
thubusercontent.com/9010790/187493350-a1e4bf3e-0423-4dfd-b495-358a1653d226.png">
<img width="320" alt="Screen Shot 2022-08-30 at 12 14 32 PM" src="https://user-images.githubusercontent.com/9010790/187493345-b6e0655b-47a0-420b-817a-29af85f0cf0b.png">
